test/WebJobs.Script.Tests.Integration/TestFunctionHost.cs

```diff
@@ -283,12 +288,34 @@ public void ConfigureServices(IServiceCollection services)
                 _postConfigure?.ConfigureServices(services);
             }
 
-            public void Configure(AspNetCore.Builder.IApplicationBuilder app, AspNetCore.Hosting.IApplicationLifetime applicationLifetime, AspNetCore.Hosting.IHostingEnvironment env, ILoggerFactory loggerFactory)
+            public void Configure(AspNetCore.Builder.IApplicationBuilder app, AspNetCore.Hosting.IApplicationLifetime applicationLifetime, AspNetCore.Hosting.IHostingEnvironment env, Microsoft.Extensions.Logging.ILoggerFactory loggerFactory)
             {
                 _startup.Configure(app, applicationLifetime, env, loggerFactory);
             }
         }
 
+        private class TestServiceProvider : IServiceProvider
```

Comment:
This shouldn't be needed... you should be able to build a new service provider with the appropriate services. There are examples of that in the test suite.